src/Microsoft.Azure.WebJobs.Host/Bindings/ExecutionContext/ExecutionContext.cs

```diff
@@ -32,5 +32,14 @@ public class ExecutionContext
         /// A host can set this via <see cref="CoreJobHostConfigurationExtensions.UseCore(JobHostConfiguration, string)"/>
         /// </remarks>
         public string FunctionAppDirectory { get; set; }
+
+        /// <summary>
+        /// Gets the function current retry count.
+        /// -1 - no retry policy is configured.
+        /// 0 - the function with retry policy runs first time. 
+        /// 1 - first retry attempt.
+        /// (n) - (n) retry attempt.
+        /// </summary>
+        public int RetryCount { get; internal set; }
```

Comment:
I wonder if people will be confused by this, given that triggers already have their own level of inherent retries. E.g. for a failed queue message, it might be invocation #3 of the function, and this retry will be 0. I.e. there are now have 2 layers of retries - from this new policy, and from the trigger itself.

Reply:
Yes, I see how this can be confusing. 
But we need the property in `ExecutionContext` to indicate retry attempt number. I think we can describe the behavior of the property in documentation and point that it only make sense for EventHub/CosmosDB/Kafka triggers.

Reply:
I'd add a cref here to RetryAttribute, basically clarifying that this retry count only applies if the retry annotation is used.

Reply:
Removed `RetryCount` from execution context.